authors/apps/profiles/test/test_user_profile.py

```diff
@@ -0,0 +1,106 @@
+from rest_framework.views import status
+from rest_framework.test import APITestCase, APIClient
+
+
+class LoginTestCase(APITestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.login_url = reverse('authentication:auth-login')
+        self.signup_url = reverse('authentication:auth-register')
+        self.data_john = {
+            "user": {
+                "username": "johndoe",
+                "email": "john@gmail.com",
+                "password": "@Qwerty12345"
+            }}
+        self.data_jane = {
+            "user": {
+                "username": "janedoe",
+                "email": "jane@gmail.com",
+                "password": "@Qwerty12345"
+            }}
+        self.profile_data = {
+            "first_name": "kwame",
+            "last_name": "asiago",
+            "gender": "F",
+            "bio": "my bio"
+        }
+
+        self.invalid_gender = {
+            "first_name": "kwame",
+            "last_name": "asiago",
+            "gender": "G",
+            "bio": "my bio"
+        }
+
+    def register_user(self, user):
+        register = self.client.post(
+            self.signup_url, user, format='json')
+        return register.data['token']
+
+    def test_get_user_profile(self):
+        """Test get profile upon registrations"""
+        self.register_user(self.data_john)
+        # get user profile
+        profile = self.client.get(
+            reverse('profiles:put-profile', kwargs={'username': self.data_john['user']['username']}), format='json')
+        self.assertEqual(profile.status_code, status.HTTP_200_OK)
+
+    def test_updating_profile(self):
+        """Test updating profile upon registrations"""
+        token = self.register_user(self.data_john)
+        # update profile
+        profile = self.client.put(reverse('profiles:put-profile',
+                                          kwargs={'username': self.data_john['user']['username']}),
+                                  self.profile_data, format='json', HTTP_AUTHORIZATION='token {}'.format(token))
+        self.assertEqual(profile.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            profile.data, {'message': 'Profile has been updated successfully'})
+
+    def test_updating_invalid_gender(self):
+        """Test updating an invalid gender"""
+        token = self.register_user(self.data_john)
+        # update profile
+        profile = self.client.put(
+            reverse(
+                'profiles:put-profile',
+                kwargs={'username': self.data_john['user']['username']}),
+            self.invalid_gender,
+            format='json',
+            HTTP_AUTHORIZATION='token {}'.format(token))
+        self.assertEqual(profile.status_code, status.HTTP_400_BAD_REQUEST)
+        msg = 'Please enter M if you are male, F if you are female or N if you do not want to disclose'
+        data = {'errors': {'gender': [msg]}}
+        self.assertEqual(data, json.loads(profile.content))
```

Comment:
Just so you know... `profile.content` is A bytestring which you have to json.loads on... but profile.data is a python object that will equal your `data` variable without json.loads

Reply:
I had tried with data but it was not returning with the correct data